demos/text_detection_demo/cpp/main.cpp

```diff
@@ -188,19 +217,19 @@ int main(int argc, char *argv[]) {
                 });
                 rects.resize(static_cast<size_t>(FLAGS_max_rect_num));
             }
-
-            int num_found = text_recognition.is_initialized() ? 0 : static_cast<int>(rects.size());
+            int num_found = text_recognition != nullptr ? 0 : static_cast<int>(rects.size());
 
             for (const auto &rect : rects) {
                 cv::Mat cropped_text;
                 std::vector<cv::Point2f> points;
                 int top_left_point_idx = 0;
 
-                if (rect.size != cv::Size2f(0, 0) && text_detection.is_initialized()) {
+                if (rect.size != cv::Size2f(0, 0) && text_detection != nullptr) {
                     std::chrono::steady_clock::time_point begin = std::chrono::steady_clock::now();
                     points = floatPointsFromRotatedRect(rect);
                     topLeftPoint(points, &top_left_point_idx);
-                    cropped_text = cropImage(image, points, text_recognition.input_size(), top_left_point_idx);
+                    auto size = text_recognition == nullptr ? cv::Size(rect.size) : text_recognition->input_size();
+                    cropped_text = cropImage(image, points, size, top_left_point_idx);
```

Comment:
You already checked `text_detection != nullptr`. You can revert this change except for `operator->` call.